src/map.cpp

```diff
@@ -413,17 +413,7 @@ bool map::vehact( vehicle &veh )
         veh.falling = false;
     }
 
-    // Mph lost per tile when coasting, by an ideal vehicle
-    int base_slowdown = veh.skidding ? 50 : 5;
-    if( should_fall ) {
-        // Just air resistance
-        base_slowdown = 1;
-    }
-
-    // "Anti-ideal" vehicle slows down up to 10 times faster than ideal one
-    const float k_slowdown = 20.0f / ( 2.0f + 9 * ( veh.k_dynamics() * veh.k_mass() ) );
-    const int slowdown = veh.drag() + (int)ceil( k_slowdown * base_slowdown );
-    add_msg( m_debug, "%s vel: %d, slowdown: %d", veh.name.c_str(), veh.velocity, slowdown );
+    const int slowdown = veh.slowdown( should_fall );
```

Comment:
Boolean arguments are best avoided whenever possible, with the exception of functions like `set_boolean_property(bool)`. I assume you could use `vehicle::falling` within the function to both get rid of the flag, and improve encapsulation (so that `slowdown` would unequivocally describe the state of the vehicle).

BTW [this line](https://github.com/CleverRaven/Cataclysm-DDA/pull/24854/files#diff-ab005ad9757130721ffb66a09833090aR394) looks suspicious. It justifies the assumption (`should_fall` can only be true if `vehicle::falling` is also true), but contradicts the above comment (which clearly states `OR`).

Reply:
I just want to refactor the slowdown code, because I want to use it later to help calculate and display effective power/fuel consumption and recharge rates. I don't want to do too much towards generally fixing weird things in the map code that I don't understand.